### PR TITLE
Add option for creating account from external/legacy details.

### PR DESCRIPTION
**Pull request by mankyKitty**

### CODE REVIEW

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs [hunk 1]
```diff
@@ -171,18 +182,56 @@ createAccountSplash model name chain mPublicKey keysetPresets = Workflow $ do
         pure $ ( constDyn $ mkAccountGuard (Set.singleton key) "keys-all"
                , emptyKeysetPresets
                )
+
+    dialogSectionHeading mempty "Transaction Gas Payer"
+    (_onCreateSelect, dCreateSelect) <- divClass "group" $ mdo
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: Unused `_onCreateSelect` - remove?

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs [hunk 2]
```diff
@@ -171,18 +182,56 @@ createAccountSplash model name chain mPublicKey keysetPresets = Workflow $ do
...
+    next <- confirmButton (def & uiButtonCfg_disabled .~ nextDisabled) "Next"
+
+    let
+      notGasPayerWF = createAccountNotGasPayer model name chain mPublicKey keysetSelections
+      gasPayerWF = createAccountConfig model name chain mPublicKey keysetSelections
+      externalPayerWF = createAccountFromExternalAccount  model name chain mPublicKey keysetSelections
+
+      nextWF = ffor2 keyset dCreationMethod $ \mkeys -> \case
+        CreateAccountMethod_ShareTxBuilder -> notGasPayerWF <$> mkeys
+        CreateAccountMethod_SelfGasPayer -> gasPayerWF <$> mkeys
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: Better rename `gasPayer` to `internalGasPayer` or inline the bindings

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs [hunk 3]
```diff
@@ -171,18 +182,56 @@ createAccountSplash model name chain mPublicKey keysetPresets = Workflow $ do
...
-    pure (cancel, next)
+
+    next <- confirmButton (def & uiButtonCfg_disabled .~ nextDisabled) "Next"
+
+    let
+      notGasPayerWF = createAccountNotGasPayer model name chain mPublicKey keysetSelections
+      gasPayerWF = createAccountConfig model name chain mPublicKey keysetSelections
+      externalPayerWF = createAccountFromExternalAccount  model name chain mPublicKey keysetSelections
+
+      nextWF = ffor2 keyset dCreationMethod $ \mkeys -> \case
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: `\mkeys -> ffor mkeys $ \case`

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs [hunk 4]
```diff
@@ -223,11 +273,70 @@ createAccountNotGasPayer ideL name chain mPublicKey selectedKeyset keyset = Work
...
+        _receiveFromLegacy_transferInfo receiveConf
+
+      onSubmit = tagMaybe (current requiredInfo) submit
+
+      rewrapWF wfFn (netI, tfrI) = mapWorkflow ("Creating account",) $
+        wfFn netI tfrI keyset
+
+    pure ( ("Sender Details" , (_receiveFromLegacy_conf receiveConf, never) )
+         , leftmost
+           [ createAccountSplash model name chain mPublicKey selectedKeyset <$ back
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: `mPublicKey selectedKeyset` are only used for rebuilding the widget we came from, better pass it in with the usual `self` binding trick.

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs (hunk 3 above)
Comment (mankyKitty) [reply frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: Why?

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs (hunk 4 above)
Comment (mankyKitty) [reply frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: This causes the form to be cleared when you click 'back'

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs (hunk 3 above)
Comment (alexfmpe) [reply frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: Ah indeed, the change you did is much better

File: frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs [hunk 5]
```diff
@@ -409,7 +393,8 @@ createAccountConfig ideL name chainId mPublicKey selectedKeyset keyset = Workflo
                (current $ ideL ^. network_selectedNodes)
                command
 
-           , createAccountSplash ideL name chainId mPublicKey selectedKeyset <$ onBack
+           -- , createAccountSplash ideL name chainId mPublicKey selectedKeyset <$ onBack
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/Dialogs/AddVanityAccount.hs]: Leftover

**Pull request merged**